src/test/java/org/scijava/io/handle/DataHandlesTest.java

```diff
@@ -0,0 +1,135 @@
+	private static EventService eventService;
+
+	@BeforeClass
+	public static void classSetup() {
+		final Context ctx = new Context(DataHandleService.class,
+			ThreadService.class, EventService.class);
+		handles = ctx.service(DataHandleService.class);
+		threadService = ctx.service(ThreadService.class);
+		eventService = ctx.service(EventService.class);
+	}
+
+	@Before
+	public void setUp() throws IOException {
+
+		final String prefix = getClass().getName();
+		inFile = File.createTempFile(prefix, "testCopyIn");
+		outFile = File.createTempFile(prefix, "testCopyOut");
+
+		data = randomBytes(0xbabebabe);
+		FileUtils.writeFile(inFile, data);
+	}
+
+	@After
+	public void tearDown() {
+		inFile.delete();
+		outFile.delete();
+	}
+
+	@AfterClass
+	public static void cleanup() {
+		handles.context().dispose();
+	}
+
+	@Test
+	public void testDefaultCopy() throws IOException {
+
+		try (DataHandle<Location> src = handles.create(new FileLocation(inFile));
+				final DataHandle<Location> dest = handles.create(new FileLocation(
+					outFile)))
+		{
+			DataHandles.copy(src, dest);
+		}
+		final byte[] result = FileUtils.readFile(outFile);
+		assertArrayEquals(data, result);
+	}
+
+	@Test
+	public void testCopyTask() throws IOException {
+		try (DataHandle<Location> src = handles.create(new FileLocation(inFile));
+				final DataHandle<Location> dest = handles.create(new FileLocation(
+					outFile)))
+		{
+			Task t = new DefaultTask(threadService, eventService);
+			DataHandles.copy(src, dest, t);
+			assertEquals(t.getProgressValue(), src.length());
+		}
+		final byte[] result = FileUtils.readFile(outFile);
+		assertArrayEquals(data, result);
+	}
+
+	@Test
+	public void testCopyLength() throws IOException {
+		int sliceSize = 50_000;
+		try (DataHandle<Location> src = handles.create(new FileLocation(inFile));
+				final DataHandle<Location> dest = handles.create(new FileLocation(
+					outFile)))
+		{
+			DataHandles.copy(src, dest, sliceSize);
+		}
+		final byte[] result = FileUtils.readFile(outFile);
+
+		byte[] expected = new byte[sliceSize];
+		System.arraycopy(data, 0, expected, 0, sliceSize);
+		assertArrayEquals(expected, result);
+	}
+
+	@Test
+	public void testCopyLengthTask() throws IOException {
+		int sliceSize = 50_000;
+		try (DataHandle<Location> src = handles.create(new FileLocation(inFile));
+				final DataHandle<Location> dest = handles.create(new FileLocation(
+					outFile)))
+		{
+			Task t = new DefaultTask(threadService, eventService);
+			DataHandles.copy(src, dest, sliceSize, t);
+			assertEquals(t.getProgressValue(), sliceSize);
+		}
+		final byte[] result = FileUtils.readFile(outFile);
+
+		byte[] expected = new byte[sliceSize];
+		System.arraycopy(data, 0, expected, 0, sliceSize);
+		assertArrayEquals(expected, result);
+	}
+
+	private byte[] randomBytes(final long seed) {
+		final MersenneTwisterFast r = new MersenneTwisterFast(seed);
+		final byte[] ldata = new byte[2938740];
```

Comment:
You use underscore in numeric constant above, but not here. Let's be consistent!